libtock/tock.c

```diff
@@ -89,12 +90,56 @@ void yield(void) {
     // registers r4-r8, r10, r11 and SP (and r9 in PCS variants that designate
     // r9 as v6) As our compilation flags mark r9 as the PIC base register, it
     // does not need to be saved. Thus we must clobber r0-3, r12, and LR
+    register uint32_t wait asm ("r0")       = 1; // yield-wait
+    register uint32_t wait_field asm ("r1") = 0; // yield result ptr
     asm volatile (
       "svc 0       \n"
       :
+      : "r" (wait), "r" (wait_field)
+      : "memory", "r2", "r3", "r12", "lr"
+      );
+  }
+}
+
+int yield_no_wait(void) {
+  if (task_cur != task_last) {
```

Comment:
why not put all this logic into a `yield_internals(int wait)` and then have `yield()` and `yield_no_wait()` just be wrappers that pass the appropriate values for the wait field?

Reply:
I've generally been trying to do minimum-change PRs, rather than go down the path of refactoring.  But sure, I can refactor here.